### PR TITLE
Add  get_state and donefile functions

### PR DESCRIPTION
**Pull request by kwaegema**
- get the state of a running version (now tested and working)
- add option to write a cachefile with some results after sync is done, for postprocessing or monitoring

### CODE REVIEW

**Comment by kwaegema** [comment]
@JensTimmerman 
WARN: No local packages or download links found for vsc-utils>=1.7.7
error: Could not find suitable distribution for Requirement.parse('vsc-utils>=1.7.7')
vsc-utils on master is 1.7.7

**Comment by JensTimmerman** [comment]
depends on https://github.com/hpcugent/vsc-utils/pull/33
the vsc-utils in master doesn't correctly build yet

**Comment by JensTimmerman** [comment]
turns out it depends on this https://github.com/hpcugent/vsc-utils/pull/34

**Comment by kwaegema** [comment]
test this please

**Comment by kwaegema** [comment]
@JensTimmerman I'm unable to trigger jenkins

**Comment by JensTimmerman** [comment]
test this please

**Comment by wdpypere** [comment]
@kwaegema ping?

**Comment by stdweird** [comment]
retest this please

**Comment by stdweird** [comment]
retest this please

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,34 @@
+*.py[co]
+
+# Packages
+*.egg
```
Comment (stdweird) [inline .gitignore]: add `.eggs`

**Comment by stdweird** [comment]
test this please

**Comment by kwaegema** [comment]
@stdweird @JensTimmerman hurray! thanks

File: lib/vsc/zk/rsync/controller.py [hunk 2]
```diff
@@ -66,17 +65,18 @@ def __init__(self, hosts, session=None, name=None, default_acl=None,
...
 
+        self.rsyncpath = rsyncpath.rstrip(os.path.sep)
+
         if not netcat:
-            rsyncpath = rsyncpath.rstrip(os.path.sep)
-            if not os.path.isdir(rsyncpath):
+            if verifypath and not self.basepath_ok():
                 self.log.raiseException('Path does not exists in filesystem: %s' % rsyncpath)
             if not os.path.isdir(self.RSDIR):
                 os.mkdir(self.RSDIR, 0700)
```
Comment (JensTimmerman) [inline lib/vsc/zk/rsync/controller.py]: make this `0o700` to make it clearer this is an octal number (required in python3)

**Comment by kwaegema** [comment]
@stdweird done

**Comment by kwaegema** [comment]
@JensTimmerman done

**Pull request merged**